go/core/action.go

```diff
@@ -28,6 +28,26 @@ import (
 	"github.com/invopop/jsonschema"
 )
 
+// Middleware is a functions that take in the request object, response object,
+// and the next middleware function for the action.
+//
+// Middleware functions can:
+//   - execute arbitrary code;
+//   - change the request and response;
+//   - terminate response by returning a response (or error);
+//   - call the next middleware function.
+//
+// If the current middleware function does not terminate the middlware chain it
+// must call next middlware in the chain to pass control to the next middleware
+// function, otherwise the request will be left hanging.
+type Middleware[I, O any] func(context.Context, I, func(ctx context.Context, input I) (O, error)) (O, error)
+
+// Middlewares returns an array of middlewares that are passes in as an argument.
+// `core.Middlewares(apple, banana)` is identical to `[]core.Middleware[InputType, OutputType]{apple, banana}`
```

Comment:
Markdown syntax doesn't work in Go doc comments.

Reply:
Sad... removed.